pkg/services/ssosettings/ssosettingsimpl/service_test.go

```diff
@@ -762,7 +798,7 @@ func TestService_Upsert(t *testing.T) {
 		reloadable := ssosettingstests.NewMockReloadable(t)
 		reloadable.On("Validate", mock.Anything, settings).Return(nil)
 		reloadable.On("Reload", mock.Anything, mock.MatchedBy(func(settings models.SSOSettings) bool {
-			wg.Done()
+			defer wg.Done()
 			return settings.Provider == provider &&
 				settings.ID == "someid" &&
 				maps.Equal(settings.Settings, map[string]any{
```

Comment:
The read of the map in `maps.Equal` was racing with a concurrent write after calling `wg.Wait()`, so I added the `defer` to make sure the read was performed before `wg.Wait()` and the subsequent write would be safe.

Reply:
Thanks!